tasks/main.yml

```diff
@@ -1,12 +1,44 @@
 ---
+
+- name: install puppetlabs repo
+  yum:
+    name: "{{item}}"
+    state: present
+    validate_certs: no
+  with_items:
+    - "{{puppetlabs_repo_url}}"
+  when: puppetize_manage_yumrepo
+
 - name: install puppet
-  yum: name='puppet' state=present
+  yum:
+    name: 'puppet'
+    state: present
   tags: ['install_puppet_client']
   when: puppet_run_only == False
 
+- name: stat /etc/puppetlabs
+  stat:
+    path: /etc/puppetlabs
+  register: reg_puppetize_stat_puppetlabs
+  ignore_errors: True
+  changed_when: False
+
+- name: set fact puppetize_version to 3 if /etc/puppetlabs does not exist
+  set_fact:
+    puppetize_version: 3
+  when: reg_puppetize_stat_puppetlabs.stat.exists == False
+
+- name: set fact puppetize_version to 4 if /etc/puppetlabs does exists
+  set_fact:
+    puppetize_version: 4
+  when: reg_puppetize_stat_puppetlabs.stat.exists
+
+- name: include vars depending on puppet version
+  include_vars:
+    file: "puppetize_vars_{{ puppetize_version }}.yml"
 
-- name: Get puppet.conf contents
-  command: grep environment /etc/puppet/puppet.conf
+- name: Check if puppet.conf contains "environment ="
+  command: "grep 'environment =' {{ puppet_etc_dir }}/puppet.conf"
```

Comment:
Personally I would allow some artistic freedom for a script or maybe even a human modifying that file to do some permutations of 

```
environment=
environment =
environment  =
```

Did you find some nice puppet4 recommendation for being this strict?

Reply:
BTW either way this is not a big deal / doesn't need a fix, just curious

Reply:
I relaxed the grep again. I did not. Another assumption "because puppet.conf looked like that" :)